src/main/java/org/opensearch/flowframework/transport/ReprovisionWorkflowTransportAction.java

```diff
@@ -452,8 +460,15 @@ private void executeWorkflow(
                                 }
                             })
                         );
+                    } else {
+                        // We've completed provisioning asynchronously
+                        TenantAwareHelper.releaseProvision(tenantId);
                     }
-                }, exception -> { logger.error("Failed to update workflow state for workflow {}", workflowId, exception); })
+                }, exception -> {
+                    // We've completed provisioning asynchronously but failed state update
+                    TenantAwareHelper.releaseProvision(tenantId);
```

Comment:
In the concurrent map should we also store the status of provisioning along with the tenantId? To keep a track for which tenantId we had a successful provisioning

Reply:
Not sure what you're asking. This is intended to be a very lightweight throttling mechanism, not a stats collector.